elasticsearch/main.tf

```diff
@@ -0,0 +1,112 @@
+locals {
+  tags = merge(
+    var.tags,
+    {
+      "Name"        = "${var.project}-${var.environment}-${var.name}"
+      "Environment" = var.environment
+      "Project"     = var.project
+    },
+  )
+
+  tags_noname = merge(
+    var.tags,
+    {
+      "Environment" = var.environment
+      "Project"     = var.project
+    },
+  )
+}
+
+data "aws_region" "current" {}
+
+resource "aws_elasticsearch_domain" "es" {
+  domain_name           = "${var.project}-${var.environment}-${var.name}"
+  elasticsearch_version = var.elasticsearch_version
+
+  cluster_config {
+    instance_count           = var.instance_count
+    instance_type            = var.instance_type
+    dedicated_master_enabled = var.dedicated_master_enabled
+    dedicated_master_count   = var.dedicated_master_enabled ? var.dedicated_master_count : 0
+    dedicated_master_type    = var.dedicated_master_enabled ? var.dedicated_master_type : ""
+    zone_awareness_enabled   = var.zone_awareness_enabled
+  }
+
+  ebs_options {
+    ebs_enabled = contains(var.ephemeral_list, var.instance_type) ? false : true
+    volume_type = contains(var.ephemeral_list, var.instance_type) ? null : var.volume_type
+    volume_size = contains(var.ephemeral_list, var.instance_type) ? null : var.volume_size
+    iops        = var.volume_type == "io1" ? var.volume_iops : null
+  }
+
+  snapshot_options {
+    automated_snapshot_start_hour = var.snapshot_start_hour
+  }
+
+  tags = local.tags
+
+  advanced_options = {
+    "rest.action.multi.allow_explicit_index" = tostring(var.options_rest_action_multi_allow_explicit_index)
+    "indices.fielddata.cache.size"           = tostring(var.options_indices_fielddata_cache_size)
+    "indices.query.bool.max_clause_count"    = tostring(var.options_indices_query_bool_max_clause_count)
+  }
+
+  encrypt_at_rest {
+    enabled = var.disable_encrypt_at_rest ? false : contains(var.encryption_list, var.instance_type)
+  }
+
+  log_publishing_options {
+    enabled                  = var.logging_enabled
+    log_type                 = "INDEX_SLOW_LOGS"
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.cwl_index.arn
+  }
+
+  log_publishing_options {
+    enabled                  = var.logging_enabled
+    log_type                 = "SEARCH_SLOW_LOGS"
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.cwl_search.arn
+  }
+
+  dynamic "vpc_options" {
+    for_each = [var.vpc_id]
```

Comment:
You'll have to produce an empty list if you want this to work. As it is now, if `var.vpc_id` is `null` the `for_each` will still be a list with one item (`[null]`).

Maybe something like this:
```
for_each = var.vpc_id == null ? [] : [var.vpc_id]
```

Reply:
right, just noticed that too